### PR TITLE
misc improvements

### PR DESCRIPTION
**Pull request by roock**
- missing dpkg options when installing fdisk and ssh
- implement shellcheck hints
- move activation of needrestart to the end of the upgrade
- adding force mode to bypass question about uninstalling packages

### CODE REVIEW

File: upgrade.sh [hunk 1]
```diff
@@ -104,7 +106,8 @@ set -x
 
 apt-get clean
 apt-get -y --purge autoremove
-while deborphan -n | grep -q . ; do echo "Deborphan remove...."; apt-get purge $(deborphan -n); done
+# shellcheck disable=SC2046
+while deborphan -n | grep -q . ; do echo "Deborphan remove...."; apt-get -y purge $(deborphan -n); done
```
Comment (mika) [inline upgrade.sh:123]: The `apt -get` invocation with `-y` now no longer asks for confirmation, is that by behavior change by intention?

File: upgrade.sh [hunk 2]
```diff
@@ -144,7 +144,7 @@ if ! test -h /bin ; then
   apt-get remove --purge -y usrmerge
 fi
 apt-get clean
-apt-get --purge autoremove
+apt-get --purge -y autoremove
```
[the remark this answers is not in the document]
Comment (roock) [reply upgrade.sh]: @mika changed the default back to the previous behavior, with the possibility to use the force mode to skip the confirmation

File: upgrade.sh [hunk 3]
```diff
@@ -93,8 +108,10 @@ set +x
 DEINSTALL_PACKAGES=$(dpkg --get-selections | awk '$2=="deinstall" {print $1}')
 if [ -n "$DEINSTALL_PACKAGES" ]; then
   echo "Some packages are to be deinstalled: ${DEINSTALL_PACKAGES}"
-  echo "really purge these [y/N]?"
-  if read -r ans && [ "$ans" = "y" ] ; then
+  $FORCE || echo "really purge these [y/N]?"
+  if $FORCE || ( read -r ans && [ "$ans" = "y" ] ) ; then
+    dpkg --purge "${DEINSTALL_PACKAGES}"
```
Comment (zeha) [inline upgrade.sh:113]: this line looks like a leftover?

Comment (roock) [reply upgrade.sh]: @zeha yeah, sorry. leftover from rebasing. fixed now

Comment (zeha) [reply upgrade.sh]: thanks!

**Comment by zeha** [comment]
The deborphan change does not actually work:

```
Deborphan remove....
++ deborphan -n
+ apt-get '' purge apt-transport-https libapt-inst2.0 libdns-export1104 libffi6 libglib2.0-data libhogweed4 libip4tc0 libip6tc0 libiptc0 libjson-c3 libldap-common libprocps7 libsasl2-modules
E: Invalid operation
+ deborphan -n
```

Thats apt being strange I guess...

**Comment by zeha** [comment]
> The deborphan change does not actually work:
> ...

should be fixed by 055f63228f35cd27ab6c6b0bf061bd7603245864

**Pull request merged**